pkg/operator/starter.go

```diff
@@ -115,33 +158,64 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 			}
 			return isSNO
 		},
-	).AddKubeInformers(kubeInformersForNamespaces)
+	).AddKubeInformers(managementKubeInformersForNamespaces)
 
-	controllerDeploymentManifest, err := assets.ReadFile("csi_controller_deployment.yaml")
+	controllerDeploymentManifest, err := namespacedAssetFunc("csi_controller_deployment.yaml")
 	if err != nil {
 		return err
 	}
+
+	var webhookHooks []validatingWebhookConfigHook
+	if isHyperShift {
+		webhookHooks = []validatingWebhookConfigHook{
+			hyperShiftSetWebhookService(),
+		}
+	}
+	webhookController := NewValidatingWebhookConfigController(
+		"WebhookController",
+		guestOperatorClient,
+		guestKubeClient,
+		guestKubeInformersForNamespaces.InformersFor("").Admissionregistration().V1().ValidatingWebhookConfigurations(),
+		namespacedAssetFunc,
+		"webhook_config.yaml",
+		controllerConfig.EventRecorder,
+		webhookHooks,
+	)
+
+	var deploymentHooks []dc.DeploymentHookFunc
+	if isHyperShift {
+		deploymentHooks = []dc.DeploymentHookFunc{
+			hyperShiftReplaceNamespaceHook(operandNamespace),
+			hyperShiftRemoveNodeSelector(),
+			hyperShiftAddKubeConfigVolume("admin-kubeconfig"), // TODO: use dedicated secret for Snapshots
+		}
+	} else {
+		// Standalone OCP
+		deploymentHooks = []dc.DeploymentHookFunc{
+			csidrivercontrollerservicecontroller.WithControlPlaneTopologyHook(guestConfigInformers),
+			csidrivercontrollerservicecontroller.WithReplicasHook(
```

Comment:
We shouldn't have any replicas in Hypershift?

Reply:
Added them to deployment asset itself, together with a big TODO item. The operator should use value from HostedCluster / HostedControlPlane CR - CPO would need to pass it as an argument?